lib/metadata/index.js

```diff
@@ -131,10 +130,26 @@ Metadata.prototype.buildTokens = function (hosts) {
   this.primaryReplicas = primaryReplicas;
   //All the tokens in ring order
   this.ring = allSorted;
+
+  // Build TokenRanges.
+  const tokenRanges = new Set();
```

Comment:
I would avoid using a `Set` as ES6 doesn't provide a way to override element comparison / hash code, so its not very useful with instances of type object.

Reply:
The more I think about, I think both `Array` / `Set` are valid return types, so it's ok for me to leave it as is.

Reply:
👍 , will stick with `Set` then.  My initial thought was it's useful as a means of communicating that would won't get duplicates (in terms of strict equality (`===`)), however there is a slight cost in that the APIs between arrays and sets are different so it could be annoying for users who want to interact between the two, but that doesn't seem too troubling.